pulsator4droid/src/main/java/pl/bclogic/pulsator4droid/library/roundedrectangle/RoundedRectanglePulsatorLayout.java

```diff
@@ -0,0 +1,167 @@
+package pl.bclogic.pulsator4droid.library.roundedrectangle;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Bitmap.Config;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.Path.Direction;
+import android.graphics.PorterDuff.Mode;
+import android.graphics.PorterDuffXfermode;
+import android.graphics.RectF;
+import android.util.AttributeSet;
+
+import pl.bclogic.pulsator4droid.library.PulsatorLayout;
+import pl.bclogic.pulsator4droid.library.PulseShape;
+
+public class RoundedRectanglePulsatorLayout extends PulsatorLayout {
+
+    private static final int ALPHA_MAX = 0x88;
+
+    private int mMaskWidth = 0;
+    private int mMaskHeight = 0;
+
+    private long mStartTime;
+    private Bitmap mBitmap;
+    private Canvas mCanvas;
+    private Paint mBitmapPaint;
+    private Paint mMaskPaint;
+
+    public RoundedRectanglePulsatorLayout(Context context) {
+        this(context, null, 0);
+    }
+
+    public RoundedRectanglePulsatorLayout(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public RoundedRectanglePulsatorLayout(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+
+        mBitmapPaint = new Paint();
+        mMaskPaint = new Paint();
+        mMaskPaint.setXfermode(new PorterDuffXfermode(Mode.CLEAR));
+    }
+
+    @Override protected void onDraw(Canvas canvas) {
+
+        RectF pulseRectangle = ((PulseRoundedRectangle) mPulseShape).getRect();
+
+        if (mMask == null
```

Comment:
this could be simplified by a convenience method better explaining the condition

Reply:
done